### PR TITLE
Feature/migrate away from mapbox

### PR DESCRIPTION
**Pull request by ngottlieb**
[Migrate Away from Mapbox for Cost Savings#2715](https://github.com/AmericanWhitewater/wh2o/issues/2715)

- migrates to maplibre gl open source library (v. 1.15 to maintain drop-in compatibility)
- introduces new ESRI api key env var
- switches to ESRI base layers for all maps *except* reach creator

Note: we may be able to transition the reach creator by using ArcGIS Hub's [NHD feature layer](https://hub.arcgis.com/maps/fedmaps::national-hydrography-dataset/about). I don't have a complete grasp on how this works, but if it works the way I think it does, it's likely to actually significantly improve our reach creator mapping interface as well as allowing us to stop using Mapbox altogether. (Update: I just found out It's also possible we could do this directly from a USGS hosted service!!! -- https://apps.nationalmap.gov/services/ lots of interesting options here https://www.usgs.gov/national-hydrography/access-national-hydrography-products)

### CODE REVIEW

**Comment by ngottlieb** [comment]
found some issues with this when I put it up on beta for testing so will need a bit more work

File: src/app/views/river-detail/__tests__/river-detail.spec.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ test.todo('revive tests')
 // import { createWrapper } from '@/utils'
```
Comment (ryangroth5) [inline src/app/views/river-detail/__tests__/river-detail.spec.js:3]: now or sometime else could we just delete this file out. There's not really a point to maintaining a test that doesn't work anymore and the mapbox->maplibre isn't really adding anything to this file..

File: src/app/views/user/components/account/__tests__/user-bookmarks.spec.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // import UserBookmarks from '../user-bookmarks.vue'
```
Comment (ryangroth5) [inline src/app/views/user/components/account/__tests__/user-bookmarks.spec.js:1]: same here.

**Review by ryangroth5 — APPROVED** [review]
see comments.

File: src/app/views/user/components/account/__tests__/user-bookmarks.spec.js (hunk 2 above)
Comment (ngottlieb) [reply src/app/views/user/components/account/__tests__/user-bookmarks.spec.js:1]: I made these changes with grep/sed, hence the changes in unused/commented files

**Pull request merged**